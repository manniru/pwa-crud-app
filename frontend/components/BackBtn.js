import PropTypes from 'prop-types';
import Link from 'next/link';

const BackBtn = (props) => (
  <Link href={{ pathname: props.path }}>
    <button aria-label='back'>&lt; Back</button>
  </Link>
);

BackBtn.propTypes = {
  path: PropTypes.string.isRequired,
};

export default BackBtn;
