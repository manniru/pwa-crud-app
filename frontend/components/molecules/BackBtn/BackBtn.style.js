import styled from 'styled-components';
import Button from '../../atoms/Button/Button';
import Left from '../../design/icons/Left/Left';

export const Buttonn = styled(Button)`
  align-items: center;
  display: flex;
  padding-left: 6px;
`;

export const Leftt = styled(Left)`
  fill: #fff;
  height: 1rem;
  margin-right: 8px;
`;
