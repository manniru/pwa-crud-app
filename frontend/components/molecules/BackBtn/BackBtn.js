import PropTypes from 'prop-types';
import Link from 'next/link';
import * as sc from './BackBtn.style';

const BackBtn = (props) => (
  <Link href={{ pathname: props.path }}>
    <div>
      <sc.Buttonn className={props.className} aria-label='back' type='primary'>
        <sc.Leftt />
        Back
      </sc.Buttonn>
    </div>
  </Link>
);

BackBtn.propTypes = {
  className: PropTypes.string,
  path: PropTypes.string.isRequired,
};

export default React.memo(BackBtn);
