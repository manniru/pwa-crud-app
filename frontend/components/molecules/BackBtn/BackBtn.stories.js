import { action } from '@storybook/addon-actions';
import allCombos from '../../../.storybook/allCombos';
import BackBtn from './BackBtn';

export default { title: 'Components|Molecules', component: BackBtn };

const data = {};

const actions = {
  onClick: action('onClick'),
};

export const backBtn = () => allCombos(BackBtn, data, actions);
