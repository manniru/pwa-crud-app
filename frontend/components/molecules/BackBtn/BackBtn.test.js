import { render, cleanup } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import BackBtn from './BackBtn';
import theme from '../../../public/styles/theme.style';

const setup = (updatedProps = {}) => {
  // define initial props
  const initProps = { path: '/' };
  const mergedProps = { ...initProps, ...updatedProps };

  // render component
  const result = render(
    <ThemeProvider theme={theme}>
      <BackBtn {...mergedProps} />
    </ThemeProvider>
  );

  // target elements
  const backBtn = () => result.queryByLabelText('back');

  return { ...result, mergedProps, backBtn };
};

describe('BackBtn', () => {
  afterEach(cleanup);

  it('matches snapshot', () => {
    const utils = setup();

    expect(utils.baseElement).toMatchSnapshot();
  });

  it('renders elements', () => {
    const utils = setup();

    expect(utils.backBtn()).toBeInTheDocument();
  });
});
