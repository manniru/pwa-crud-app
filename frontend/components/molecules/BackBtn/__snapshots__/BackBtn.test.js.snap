// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackBtn matches snapshot 1`] = `
.c0 {
  border-radius: 4px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 6px;
}

.c2 {
  fill: #fff;
  height: 1rem;
  margin-right: 8px;
}

<body>
  <div>
    <div>
      <button
        aria-label="back"
        class="ant-btn c0 c1 ant-btn-primary"
        type="button"
      >
        <svg
          class="c2"
          viewBox="0 0 492 492"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M198.608 246.104L382.664 62.04c5.068-5.056 7.856-11.816 7.856-19.024 0-7.212-2.788-13.968-7.856-19.032l-16.128-16.12C361.476 2.792 354.712 0 347.504 0s-13.964 2.792-19.028 7.864L109.328 227.008c-5.084 5.08-7.868 11.868-7.848 19.084-.02 7.248 2.76 14.028 7.848 19.112l218.944 218.932c5.064 5.072 11.82 7.864 19.032 7.864 7.208 0 13.964-2.792 19.032-7.864l16.124-16.12c10.492-10.492 10.492-27.572 0-38.06L198.608 246.104z"
          />
        </svg>
        <span>
          Back
        </span>
      </button>
    </div>
  </div>
</body>
`;
